joblib/func_inspect.py

```diff
@@ -128,10 +128,10 @@ def get_func_name(func, resolv_alias=True, win_characters=True):
                 # will change with every new kernel instance. This hack
                 # always returns the same filename
                 parts[-1] = '__ipython-input__'
-            filename = '-'.join(parts)
+            filename = os.path.join(*parts)
             if filename.endswith('.py'):
                 filename = filename[:-3]
-            module = module + '-' + filename
+            module = module + filename
```

Comment:
Why not `os.path.join(module, filename)`?

Also you can tidy the `parts` stuff, I don't think you need to split and then join again. In the IPython case, filename is `'ipython-input...'`.